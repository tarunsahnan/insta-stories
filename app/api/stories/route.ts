import { NextResponse } from "next/server";
import { stories } from "./stories";
import { NextApiRequest, NextApiResponse } from "next";
import { StoryInterface } from "@/types/StoryInterface";

export async function GET() {
  return NextResponse.json({ stories }, { status: 200 });
}
