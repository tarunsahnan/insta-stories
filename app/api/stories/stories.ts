import { StoryInterface } from "@/types/StoryInterface";

export const stories: StoryInterface[] = [
  {
    userName: "Tarun Sahnan",
    userImage:
      "https://www.wikihow.com/images/thumb/9/90/What_type_of_person_are_you_quiz_pic.png/728px-What_type_of_person_are_you_quiz_pic.png",
    userStory:
      "https://img.freepik.com/free-photo/autumn-leaf-falling-revealing-intricate-leaf-vein-generated-by-ai_188544-9869.jpg?t=st=1715176859~exp=1715180459~hmac=968e24d5f6315f8618e1d0a655f47b7461d4d73cb19ac79ddbd5ddf19fc51813&w=900",
  },
  {
    userName: "Tarun Sahnan",
    userImage:
      "https://www.wikihow.com/images/thumb/9/90/What_type_of_person_are_you_quiz_pic.png/728px-What_type_of_person_are_you_quiz_pic.png",
    userStory:
      "https://img.freepik.com/free-photo/autumn-leaf-falling-revealing-intricate-leaf-vein-generated-by-ai_188544-9869.jpg?t=st=1715176859~exp=1715180459~hmac=968e24d5f6315f8618e1d0a655f47b7461d4d73cb19ac79ddbd5ddf19fc51813&w=900",
  },
  {
    userName: "Tarun Sahnan",
    userImage:
      "https://www.wikihow.com/images/thumb/9/90/What_type_of_person_are_you_quiz_pic.png/728px-What_type_of_person_are_you_quiz_pic.png",
    userStory:
      "https://img.freepik.com/free-photo/autumn-leaf-falling-revealing-intricate-leaf-vein-generated-by-ai_188544-9869.jpg?t=st=1715176859~exp=1715180459~hmac=968e24d5f6315f8618e1d0a655f47b7461d4d73cb19ac79ddbd5ddf19fc51813&w=900",
  },
  {
    userName: "Tarun Sahnan",
    userImage:
      "https://www.wikihow.com/images/thumb/9/90/What_type_of_person_are_you_quiz_pic.png/728px-What_type_of_person_are_you_quiz_pic.png",
    userStory:
      "https://img.freepik.com/free-photo/autumn-leaf-falling-revealing-intricate-leaf-vein-generated-by-ai_188544-9869.jpg?t=st=1715176859~exp=1715180459~hmac=968e24d5f6315f8618e1d0a655f47b7461d4d73cb19ac79ddbd5ddf19fc51813&w=900",
  },
  {
    userName: "Tarun Sahnan",
    userImage:
      "https://www.wikihow.com/images/thumb/9/90/What_type_of_person_are_you_quiz_pic.png/728px-What_type_of_person_are_you_quiz_pic.png",
    userStory:
      "https://img.freepik.com/free-photo/autumn-leaf-falling-revealing-intricate-leaf-vein-generated-by-ai_188544-9869.jpg?t=st=1715176859~exp=1715180459~hmac=968e24d5f6315f8618e1d0a655f47b7461d4d73cb19ac79ddbd5ddf19fc51813&w=900",
  },
  {
    userName: "Tarun Sahnan",
    userImage:
      "https://www.wikihow.com/images/thumb/9/90/What_type_of_person_are_you_quiz_pic.png/728px-What_type_of_person_are_you_quiz_pic.png",
    userStory:
      "https://img.freepik.com/free-photo/autumn-leaf-falling-revealing-intricate-leaf-vein-generated-by-ai_188544-9869.jpg?t=st=1715176859~exp=1715180459~hmac=968e24d5f6315f8618e1d0a655f47b7461d4d73cb19ac79ddbd5ddf19fc51813&w=900",
  },
  {
    userName: "Tarun Sahnan",
    userImage:
      "https://www.wikihow.com/images/thumb/9/90/What_type_of_person_are_you_quiz_pic.png/728px-What_type_of_person_are_you_quiz_pic.png",
    userStory:
      "https://img.freepik.com/free-photo/autumn-leaf-falling-revealing-intricate-leaf-vein-generated-by-ai_188544-9869.jpg?t=st=1715176859~exp=1715180459~hmac=968e24d5f6315f8618e1d0a655f47b7461d4d73cb19ac79ddbd5ddf19fc51813&w=900",
  },
  {
    userName: "Tarun Sahnan",
    userImage:
      "https://www.wikihow.com/images/thumb/9/90/What_type_of_person_are_you_quiz_pic.png/728px-What_type_of_person_are_you_quiz_pic.png",
    userStory:
      "https://img.freepik.com/free-photo/autumn-leaf-falling-revealing-intricate-leaf-vein-generated-by-ai_188544-9869.jpg?t=st=1715176859~exp=1715180459~hmac=968e24d5f6315f8618e1d0a655f47b7461d4d73cb19ac79ddbd5ddf19fc51813&w=900",
  },
];
