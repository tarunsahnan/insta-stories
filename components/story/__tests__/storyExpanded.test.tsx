import { describe, expect, test, vi } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import StoryExpanded from "../storyExpanded";
import { stories } from "@/app/api/stories/stories";

describe("StoryExpanded component", () => {
  const data = [stories[0], stories[1]];

  test("renders without crashing", () => {
    render(
      <StoryExpanded
        data={data}
        currStoryIndex={0}
        setSelectedStoryIndex={() => {}}
      />
    );
  });

  test("calls setSelectedStoryIndex when close button is clicked", () => {
    const setSelectedStoryIndexMock = vi.fn();
    const { container } = render(
      <StoryExpanded
        data={data}
        currStoryIndex={0}
        setSelectedStoryIndex={setSelectedStoryIndexMock}
      />
    );
    const closeButton = screen.getAllByTestId("close-button")[1];
    fireEvent.click(closeButton);
    expect(setSelectedStoryIndexMock).toHaveBeenCalled();
  });

  test("navigates to the next slide when clicked on right side of the screen", async () => {
    const { container } = render(
      <StoryExpanded
        data={data}
        currStoryIndex={0}
        setSelectedStoryIndex={() => {}}
      />
    );
    const component = screen.getAllByTestId("story-expanded-main")[2];
    if (!component) return;

    console.log({ component });

    fireEvent.click(component, { clientX: 300 });

    await screen.findAllByText(stories[1].userName);
  });

  test("navigates to the previous slide when clicked on left side of the screen", async () => {
    const { container } = render(
      <StoryExpanded
        data={data}
        currStoryIndex={1}
        setSelectedStoryIndex={() => {}}
      />
    );
    const component = screen.getAllByTestId("story-expanded-main")[3];

    if (!component) return;

    fireEvent.click(component, { clientX: 50 });

    await screen.findAllByText(stories[0].userName);
  });
});
