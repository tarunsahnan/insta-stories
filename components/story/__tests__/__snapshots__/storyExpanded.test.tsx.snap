// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StoryExpanded component > renders without crashing 1`] = `
<div>
  <div
    class="bg-black absolute top-0 w-full h-full overflow-hidden"
    data-testid="story-expanded-main"
  >
    <div
      class="swiper swiper-cube swiper-3d swiper-initialized swiper-horizontal swiper-watch-progress h-full w-full"
    >
      <div
        class="swiper-wrapper"
        style="transition-duration: 0ms; transition-delay: 0ms; transform-origin: 50% 50% -NaNpx; transform: translate3d(0px,0,0px) rotateX(0deg) rotateY(0deg); --swiper-cube-translate-z: 0px;"
      >
        <div
          class="swiper-slide bg-black"
        >
          <div
            class="flex justify-center items-center w-full h-full"
          >
            <img
              alt="Tarun Sahnan"
              class="w-full max-h-full object-cover"
              data-nimg="1"
              decoding="async"
              height="140"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1715178160659-7cc8ef44ade1%3Fw%3D1000%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHx0b3BpYy1mZWVkfDF8Qm4tRGpyY0Jyd298fGVufDB8fHx8fA%253D%253D&w=384&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1715178160659-7cc8ef44ade1%3Fw%3D1000%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHx0b3BpYy1mZWVkfDF8Qm4tRGpyY0Jyd298fGVufDB8fHx8fA%253D%253D&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1715178160659-7cc8ef44ade1%3Fw%3D1000%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHx0b3BpYy1mZWVkfDF8Qm4tRGpyY0Jyd298fGVufDB8fHx8fA%253D%253D&w=384&q=75 2x"
              style="color: transparent;"
              width="140"
            />
            <div
              class="flex justify-between items-center  text-white pl-3 pt-3 absolute top-0 w-full"
            >
              <div
                class="flex text-white items-center gap-3"
              >
                <img
                  alt="userImage"
                  class="w-10 h-10 rounded-full border border-white p-[1px]"
                  data-nimg="1"
                  decoding="async"
                  height="15"
                  loading="eager"
                  src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1517841905240-472988babdf9%3Fw%3D1000%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8MTZ8fHVzZXIlMjBwcm9maWxlfGVufDB8fDB8fHww&w=32&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1517841905240-472988babdf9%3Fw%3D1000%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8MTZ8fHVzZXIlMjBwcm9maWxlfGVufDB8fDB8fHww&w=16&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1517841905240-472988babdf9%3Fw%3D1000%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8MTZ8fHVzZXIlMjBwcm9maWxlfGVufDB8fDB8fHww&w=32&q=75 2x"
                  style="color: transparent;"
                  width="15"
                />
                <div>
                  <p>
                    Tarun Sahnan
                    <span
                      class="text-gray-400 text-sm"
                    >
                       5h
                    </span>
                  </p>
                  <div
                    class="flex items-center gap-1"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M512 144v288c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V144c0-26.5 21.5-48 48-48h88l12.3-32.9c7-18.7 24.9-31.1 44.9-31.1h125.5c20 0 37.9 12.4 44.9 31.1L376 96h88c26.5 0 48 21.5 48 48zM376 288c0-66.2-53.8-120-120-120s-120 53.8-120 120 53.8 120 120 120 120-53.8 120-120zm-32 0c0 48.5-39.5 88-88 88s-88-39.5-88-88 39.5-88 88-88 88 39.5 88 88z"
                      />
                    </svg>
                    <p
                      class="text-gray-300 text-sm"
                    >
                      One Take 
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="flex items-center"
              >
                <svg
                  fill="currentColor"
                  height="56"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="56"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="swiper-slide bg-black"
        >
          <div
            class="flex justify-center items-center w-full h-full"
          >
            <img
              alt="Alice Johnson"
              class="w-full max-h-full object-cover"
              data-nimg="1"
              decoding="async"
              height="140"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1715178160659-7cc8ef44ade1%3Fw%3D1000%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHx0b3BpYy1mZWVkfDF8Qm4tRGpyY0Jyd298fGVufDB8fHx8fA%253D%253D&w=384&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1715178160659-7cc8ef44ade1%3Fw%3D1000%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHx0b3BpYy1mZWVkfDF8Qm4tRGpyY0Jyd298fGVufDB8fHx8fA%253D%253D&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1715178160659-7cc8ef44ade1%3Fw%3D1000%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHx0b3BpYy1mZWVkfDF8Qm4tRGpyY0Jyd298fGVufDB8fHx8fA%253D%253D&w=384&q=75 2x"
              style="color: transparent;"
              width="140"
            />
            <div
              class="flex justify-between items-center  text-white pl-3 pt-3 absolute top-0 w-full"
            >
              <div
                class="flex text-white items-center gap-3"
              >
                <img
                  alt="userImage"
                  class="w-10 h-10 rounded-full border border-white p-[1px]"
                  data-nimg="1"
                  decoding="async"
                  height="15"
                  loading="eager"
                  src="/_next/image?url=https%3A%2F%2Fplus.unsplash.com%2Fpremium_photo-1670905033804-a08d6d73d595%3Fq%3D80%26w%3D387%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=32&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fplus.unsplash.com%2Fpremium_photo-1670905033804-a08d6d73d595%3Fq%3D80%26w%3D387%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=16&q=75 1x, /_next/image?url=https%3A%2F%2Fplus.unsplash.com%2Fpremium_photo-1670905033804-a08d6d73d595%3Fq%3D80%26w%3D387%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=32&q=75 2x"
                  style="color: transparent;"
                  width="15"
                />
                <div>
                  <p>
                    Alice Johnson
                    <span
                      class="text-gray-400 text-sm"
                    >
                       5h
                    </span>
                  </p>
                  <div
                    class="flex items-center gap-1"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M512 144v288c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V144c0-26.5 21.5-48 48-48h88l12.3-32.9c7-18.7 24.9-31.1 44.9-31.1h125.5c20 0 37.9 12.4 44.9 31.1L376 96h88c26.5 0 48 21.5 48 48zM376 288c0-66.2-53.8-120-120-120s-120 53.8-120 120 53.8 120 120 120 120-53.8 120-120zm-32 0c0 48.5-39.5 88-88 88s-88-39.5-88-88 39.5-88 88-88 88 39.5 88 88z"
                      />
                    </svg>
                    <p
                      class="text-gray-300 text-sm"
                    >
                      One Take 
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="flex items-center"
              >
                <svg
                  fill="currentColor"
                  height="56"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="56"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="swiper-cube-shadow"
          style="transform: translate3d(0px, NaNpx, NaNpx) rotateX(89.99deg) rotateZ(0deg) scale(0.94);"
        />
      </div>
    </div>
    <div
      class="absolute top-0 h-[56px] w-[56px] right-0 mt-3 z-10"
      data-testid="close-button"
    />
  </div>
</div>
`;
